<launch>
    <!-- Name of the Gazebo world -->
    <arg name="world_name"                  default="turtlebot3_world"/>

    <!--Name of the topic where the scan is getting published -->
    <arg name="scan_topic"                  default="scan"/>

    <!-- Minimum distance to the front to detect obstacles at -->
    <arg name="forward_distance"            default="0.7"/>

    <!-- Minimum distance to the sides to detect obstacles at -->
    <arg name="side_distance"               default="0.6"/>
    
    <!-- Forward speed of the robot -->
    <arg name="forward_speed"               default="0.4"/>

    <!-- Angular speed of the robot -->
    <arg name="angular_speed"               default="1.3"/>

    <!-- Launch Gazebo world -->
    <include file="$(find turtlebot3_gazebo)/launch/$(arg world_name).launch" />

    <!-- Initialize lidar saftey curtain -->
    <node name="drive_turtlebot3" pkg="obstacle_avoidance" type="drive_robot.py" output="screen">
        <remap from="scan_topic"            to="$(arg scan_topic)"/>
        <param name="forward_distance"      value="$(arg forward_distance)"/>
        <param name="side_distance"         value="$(arg side_distance)"/>
        <param name="forward_speed"         value="$(arg forward_speed)"/>
        <param name="angular_speed"         value="$(arg angular_speed)"/>
    </node>

    <!-- Visualize simulation data in RViz -->
    <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find obstacle_avoidance)/rviz/drive_turtlebot.rviz" /> -->

</launch>